packages/legend-application-query/src/stores/QueryEditorStore.ts

```diff
@@ -463,6 +726,30 @@ export abstract class QueryEditorStore {
     }
   }
 
+  *searchExistingQueryName(searchText: string): GeneratorFn<void> {
+    const isValidSearchString =
+      searchText.length >= DEFAULT_TYPEAHEAD_SEARCH_MINIMUM_SEARCH_LENGTH;
+    try {
+      const searchSpecification = new QuerySearchSpecification();
```

Comment:
I think this could become a bug, we should change the query API to have a flag to search-exact for query name, the problem can arise in the following scenarios:

1. You have 2 few queries: `MyQuery 1`, `MyQuery 12`, `MyQuery 13`
2. You search for `MyQuery 1`, we got `MyQuery 12` for some reason
3. You got wrong check because `MyQuery 12 !== MyQuery 1`

Reply:
We can do this later though, but we probably should do it.

Reply:
Makes sense 👍  https://github.com/finos/legend-engine/pull/1669